Dockerfile

```diff
@@ -54,16 +54,16 @@ RUN apt-get update \
     gcc \
     git \
     pkg-config \
-    python3.9-dev \
+    python3.12-dev \
+    python3.12-venv \
     python3-pip \
     libmysqlclient-dev \
   && rm -rf /var/lib/apt/lists/*
 
 COPY /requirements/ /requirements/
-RUN python3.9 -m pip install pip==24.0 virtualenv==20.25.1 && \
```

Comment:
Python in Ubuntu 24.04 started complaining if you install packages into the operating system using pip, because you are supposed to install them using apt-get, and use virtualenv for everything else. I was doing that before because we had terribly outdated pip otherwise, but now we caught up with Python 3.12 so we can just use builtin venv from apt-get to create the virtualenv.